share/eds/p2p/exchange_test.go

```diff
@@ -0,0 +1,110 @@
+package p2p
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	libhost "github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds"
+)
+
+func TestExchange_RequestEDS(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	store, client, server := makeExchange(t)
+
+	err := store.Start(ctx)
+	require.NoError(t, err)
+
+	err = server.Start(ctx)
+	require.NoError(t, err)
+
+	// Testcase: EDS is immediately available
+	t.Run("EDS_Available", func(t *testing.T) {
+		eds := share.RandEDS(t, 4)
+		dah := da.NewDataAvailabilityHeader(eds)
+		err = store.Put(ctx, dah.Hash(), eds)
+		require.NoError(t, err)
+
+		requestedEDS, err := client.RequestEDS(ctx, dah.Hash(), []peer.ID{server.host.ID()})
+		assert.Nil(t, err)
+		assert.Equal(t, eds.Flattened(), requestedEDS.Flattened())
+
+	})
+
+	// Testcase: EDS is unavailable initially, but is found after multiple requests
+	t.Run("EDS_AvailableAfterDelay", func(t *testing.T) {
+		storageDelay := time.Second
+		eds := share.RandEDS(t, 4)
+		dah := da.NewDataAvailabilityHeader(eds)
+		go func() {
+			time.Sleep(storageDelay)
+			err = store.Put(ctx, dah.Hash(), eds)
+			// require.NoError(t, err)
+		}()
+
+		now := time.Now()
+		requestedEDS, err := client.RequestEDS(ctx, dah.Hash(), []peer.ID{server.host.ID()})
+		finished := time.Now()
+
+		assert.Greater(t, finished.Sub(now), storageDelay)
+		assert.Nil(t, err)
```

Comment:
assert.NoError assuming you're testing to ensure that there is no error.

Reply:
done